File: LeetCode-Java/Longest Common Prefix.java
public class Solution {
    public String longestCommonPrefix(String[] strs) {
        if(strs == null || strs.length == 0)
            return new String("");
        int start = 0, end = strs[0].length();
        while(start < end)
        {
            int mid = (start + end) / 2;
            if(isCommonPrefix(strs, strs[0].substring(0, mid + 1)))
            {
                start = mid + 1;
            }
            else
            {
                end = mid;
            }
        }
        return strs[0].substring(0, start);
        
    }
    public boolean isCommonPrefix(String[] strs, String val)
    {
        for(int i = 0;i < strs.length;i ++)
        {
            if(!strs[i].startsWith(val))
                return false;
        }
        return true;
    }
}

File: Leetcode-cpp/longest_common_prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string result = "";
        if (strs.size() == 0) return result;
        int index = 0;
        while (true) {
            bool valid = true;
            for (int i = 0; i < strs.size(); ++i) {
                if (strs[i].length() == index || strs[i][index] != strs[0][index]) {
                    valid = false;
                    break;
                }
            }
            if (!valid) break;
            result += strs[0][index++];
        }
        return result;
    }
};